File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string common(string s,string r){
        int l = s.length();
        int q = r.length();
        int i=0;
        string p = "";
        while(l!=0 && q!=0){
            if(s[i]==r[i]){
                p+=s[i++];
            }
            l--;
            q--;
        }
        return p;
    }
    string longestCommonPrefix(vector<string>& strs) {
        string s = strs[0];
        for(int i=1;i<strs.size();i++){
            s = common(s,strs[i]);
        }
        return s;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String compare(String p,String q){
        int n=p.length();
        int m=q.length();
        int i=0;
        String cm="";
        while(m>0 && n>0){
            if(p.charAt(i)!=q.charAt(i)) {
                return cm;
            }
            else{
                cm+=p.charAt(i);
                i++;
            }
            m--;
            n--;
        }
        return cm;
    }
    public String longestCommonPrefix(String[] strs) {
        // if(strs[0].charAt(0)!=strs[1].charAt(0)) return "";
        String s=strs[0];
        for(int i=1;i<strs.length;i++){
            s=compare(s,strs[i]);
        }
        return s;
        
    }
}